week3/homework/src/index.js

```diff
@@ -17,58 +17,76 @@ const PORT = 3000;
 
 app.use(express.json());
 
-app.get('/', (request, response) => {
-  readAndParse('help')
-    .then(help => response.json(help))
-    .catch(error => response.status(error.statusCode || 500).json(handleError(error)));
+app.get('/', async (request, response) => {
```

Comment:
there is a trick to avoid writing try catch all the time:
```
/**
* function that takes another (async) function as a parameter
* then wraps it in a try catch
* obviously the name is arbitrary
* we only need to define this once
**/
const wrap = function(fn) { 
    return async function(req, res, next) {
             try { 
                 res.json(await fn(req, res, next));
            }
            catch(error) {
                 response.status(error.statusCode || 500).json(handleError(error);
            }
     };
}

//then when defining the handlers we use the function to wrap them
// ta dam one liner
app.get('/', wrap(async (request, response) => await readAndParse('help') );
```